File: Android/TeeTime/TeeTimeFactory.java
package com.adamwilson.golf.TeeTime;

/**
 * Created by adam on 12/30/15.
 */
public class TeeTimeFactory {
    public TeeTime createTeeTime(String courseName){
        TeeTime newTeeTime = null;

        if (courseName.equalsIgnoreCase("Irish") ||
                courseName.equalsIgnoreCase("Wolverine") ||
                courseName.equalsIgnoreCase("Buckeye")) {
            newTeeTime = new TeeTime("Bedfordhills");
        }

        return newTeeTime;
    }
}

File: IOS/TeeTimeFactory.m
//
//  TeeTimeFactory.m
//  Golf
//
//  Created by Adam Wilson on 12/29/15.
//  Copyright © 2015 Adam Wilson. All rights reserved.
//

#import "TeeTimeFactory.h"

@implementation TeeTimeFactory


-(TeeTime *)createTeeTime:(NSString *)courseName
{
    TeeTime *newTeeTime;
    
    if ([courseName isEqualToString:@"Buckeye"] ||
        [courseName isEqualToString:@"Wolverine"] ||
        [courseName isEqualToString:@"Irish"]) {
        newTeeTime = [[TeeTime alloc] initForBedford];
    }
    
    return newTeeTime;
}
@end
